skopt/space/space.py

```diff
@@ -272,9 +272,9 @@ def __init__(self, low, high, prior="uniform", base=10, transform=None,
             raise ValueError("dtype must be 'float', 'float16', 'float32'"
                              "or 'float64'"
                              " got {}".format(self.dtype))
-        elif isinstance(self.dtype, type) and self.dtype\
-                not in [float, np.float, np.float16, np.float32, np.float64]:
-            raise ValueError("dtype must be float, np.float"
+        elif isinstance(self.dtype, type) and \
+                np.issubdtype(type(self.dtype), np.floating):
+            raise ValueError("dtype must be float, np.float64"
```

Comment:
To match the previous check, the condition should probably be negated, though. :smile:

Can you also amend the error string?

Reply:
yeah sure, sorry for the long back and forth!